sulong/projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/intrinsics/multithreading/UtilAccess.java

```diff
@@ -0,0 +1,55 @@
+package com.oracle.truffle.llvm.nodes.intrinsics.multithreading;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.llvm.runtime.pointer.LLVMPointer;
+
+import java.util.Collection;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+public class UtilAccess {
```

Comment:
This can be written in a much more compact way using generics:
```
@TruffleBoundary
public static <K, V> void put(ConcurrentMap<K, V> c, K key, V value) {
    c.put(key, value);
}
```

Same thing for the get and remove methods.

Reply:
oh, yeah right. thanks for the tip.